types.go

```diff
@@ -15,54 +15,62 @@ import (
 
 const defaultGradientAbsTol = 1e-6
 
-// EvaluationType is used by a Method to specify the objective-function
-// information needed at an x location.
-type EvaluationType uint
+// Operation represents the set of operations requested by Method at each
+// iteration. It is a bitmap of *Iteration and *Evaluation constants.
+type Operation uint64
 
-// Evaluation types can be composed together using the binary or operator, for
-// example 'FuncEvaluation | GradEvaluation' to evaluate both the function
-// value and the gradient.
+// Supported Operations. Individual operations must NOT be combined together by
```

Comment:
I moved the rules to the Operation comment, but I think they would be visible through godoc even if they stayed here.